### DISCUSSION TITLE
Wie mit Systemvariablen umgehen?

### DISCUSSION THREAD

**Discussion by SukramJ** [SukramJ/aiohomematic#154] [Q&A] [answered]
Moin,
aktuell können Systemvariablen entweder, wie bei der alten Integration, als Attribute der Hub-Entität angezeigt werden, oder optional als Sensoren. Bei den Sensoren wird noch der Typ bestimmt (bool, float, int, str), aber eine Einheit wird nicht systemseitig mitgeliefert, so dass als Standardeinheit ein # bei numerischen Werten mitgegeben wird.
Grundsätzlich  werden Systemvariablen nicht über Callbacks aktualisiert, sondern müssen in Intervallen abgerufen werden.
Weitere Korrekturen wie (Name, entity_id, icon, Einheit) können natürlich nachträglich in HA angepasst.
Der schreibende Zugriff auf Systemvariablen ist natürlich weiterhin über den Service `hahm.set_variable_value` möglich.

Jetzt habe ich natürlich noch ein paar Fragen:
1. Sollen Systemvariablen als Sensor darstellbar sein?
2. Muss die Darstellung optional sein, oder sollen Systemvariablen immer als Sensor angezeigt werden?
3. Sollen Systemvariablen als Sensoren standardmäßig deaktiviert sein?

**Comment by danielperna84** [comment, marked as answer]
1. Wenn du damit das meinst wie es gerade ist, dann ja. Find' ich gut so.
2. Bools könnte man zu `binary_sensor` ändern. Ansonsten sehe ich da nix was man sonst noch machen müsste. Werte zuweisen geht ja mit dem Service.
3. Ja. Ich persönlich habe zumindest bisher nur den Sensor für den automatischen reconnect gebraucht. Das entfällt ja nun, weshalb ich die zusätzlichen Entities eigentlich nicht mehr brauchen würde.

**Reply by SukramJ** [reply]
Frage 2 bezieht sich eigentlich darauf, ob nur noch Sensoren angeboten werden sollten. 
Auf Bool als BinarySensoren darzustellen bin ich irgendwie nicht gekommen.

**Reply by danielperna84** [reply]
Im Gegensatz zu was? Mein Hub-Entity hat die Variablen ja nicht mehr als Attribute. Finde ich richtig so jetzt wo es die Sensoren gibt.

**Reply by SukramJ** [reply]
Wenn du die Sensor-Option angeklickt hast gibt es nur noch die Sensoren anstelle von Attributen.
Ich schmeisse die Option aus dem ConfigFlow raus.

**Comment by Lemocin** [comment]
> 1. Sollen Systemvariablen als Sensor darstellbar sein?

Finde das aktuelle System um einiges angenehmer als vorher. Ich kann nun in Automations einfach die Sensor-Entität(ich denke  die meinst du) anklicken, was die Sache finde ich viel angenehmer ist. Finde ich gut wenn das so bleibt.

> 2. Muss die Darstellung optional sein, oder sollen Systemvariablen immer als Sensor angezeigt werden?

Für mich reicht ein immer, wer sie nicht braucht kann sie ja deaktivieren. Aber ich persönlich habe ich nicht viele Variablen, und benutze diese auch nur um Geräte wie fremde Türsensoren an die CCU zu schleifen. Somit ists bei mir ohnehin sehr überschaubar. 

> 3. Sollen Systemvariablen als Sensoren standardmäßig deaktiviert sein?

Da ich die Variablen nur für eine einfachere Interaktion mit HA erstellt hatte, können sie von mir aus gerne aktiviert sein. Stört mich allerdings auch nicht, sie von Hand zu aktivieren.

**Reply by JoMass** [reply]
Bitte als Einheit nicht ein '#' verwenden sondern wenn diese tatsächlich von der CCU nicht mitgeliefert wird, frei lassen.  Grund: In Homeassistant wird in der Ansicht "Verlauf"  nach der Einheit gruppiert. also je eine Graphik für %, mW, A,V..... und nun auch #.  Bei vielen Variablen (bei mir ca.200 macht das eine völlig sinnlose Gruppierung die nichts aussagt. Grund für diese hohe Anzahl sind die benutzten Redmatic Flows wo CCU Variablen als Zwischenspeicher dienen, die auch einen Reboot des Systems überstehen sollen.  Danke JoMass

**Reply by SukramJ** [reply]
Dann vergib doch per customize.yml eigene Einheiten.
Die mitgelieferten Einheiten können aufgrund der Kompatibilität mit Homegear nicht verwendet werden.

**Reply by SukramJ** [reply]
Mit [customize_glob](https://www.home-assistant.io/docs/configuration/customizing-devices#manual-customization) kannst du auch Wildcards benutzen.

**Reply by JoMass** [reply]
Danke den Hinweis mit customize_glob werde ich testen. Wichtig ist mir, dass eine Lösung auch künftig beim Anlegen neuer Variablen automatisch richtig in Homeassistant ankommt und dort keiner Nacharbeit bedarf. 
Kann man das # nicht einfach weglassen oder wird das für etwas bestimmtes benötigt?

**Reply by JoMass** [reply]
customize_glob:
    "sensor.hm_sv_*":
     unit_of_measurement: ""

ist für mich eine Lösung!  Danke für den Hinweis !
